pynucastro/rates/rate.py

```diff
@@ -1566,11 +1566,12 @@ def function_string_py(self):
         fstring += f"def {self.fname}(rate_eval, T, rhoY):\n"
         fstring += f"    # {self.rid}\n"
 
-        # find the nearest value of T and rhoY in the data table
-        fstring += f"    T_nearest = ({self.fname}_data[:, TableIndex.T.value])[np.abs((10.0**{self.fname}_data[:, TableIndex.T.value]) - T).argmin()]\n"
-        fstring += f"    rhoY_nearest = ({self.fname}_data[:, TableIndex.RHOY.value])[np.abs((10.0**{self.fname}_data[:, TableIndex.RHOY.value]) - rhoY).argmin()]\n"
-        fstring += f"    inde = np.where(({self.fname}_data[:, TableIndex.T.value] == T_nearest) & ({self.fname}_data[:, TableIndex.RHOY.value] == rhoY_nearest))[0][0]\n"
-        fstring += f"    rate_eval.{self.fname} = 10.0**({self.fname}_data[inde][TableIndex.RATE.value])\n\n"
+        fstring += f"    {self.fname}_interpolator = TableInterpolator({self.fname}_info[0],\n"
+        fstring += f"                                                  {self.fname}_info[1],\n"
+        fstring += f"                                                  {self.fname}_info[2])\n"
```

Comment:
Maybe simpler with argument unpacking:
```suggestion
        fstring += f"    {self.fname}_interpolator = TableInterpolator(*{self.fname}_info)\n"
```